docs/dev/Pagination-v2.md

```diff
@@ -100,10 +111,381 @@ The discussion below uses *under max_result_window* to refer to scenarios that c
+
+```mermaid
+classDiagram
+  direction LR
+  class QueryPlan {
+    <<AbstractPlan>>
+    -Optional~int~ pageSize
+    -UnresolvedPlan plan
+    -QueryService queryService
+  }
+  class UnresolvedPlanTree {
+    <<UnresolvedPlan>>
+  }
+  QueryPlan --* UnresolvedPlanTree
+```
+
+Abstract plan tree for Initial Query Request has following changes:
+1. New Plan node -- `Paginate` -- added into the tree.
+2. `pageSize` parameter in `QueryPlan` is set, and `Paginate` is being added. It is converted to `LogicalPaginate` later.
+
+```mermaid
+classDiagram
+  direction LR
+  class QueryPlan {
+    <<AbstractPlan>>
+    -Optional~int~ pageSize
+    -UnresolvedPlan plan
+    -QueryService queryService
+  }
+  class Paginate {
+    <<UnresolvedPlan>>
+    -int pageSize
+    -UnresolvedPlan child
+  }
+  class UnresolvedPlanTree {
+    <<UnresolvedPlan>>
+  }
+  QueryPlan --* Paginate
+  Paginate --* UnresolvedPlanTree
+```
+
+Non-paging requests have the same plan tree, but `pageSize` value in `QueryPlan` is unset.
+
+Abstract plan tree for Subsequent Query Request (for second and further pages) contains only one node -- `ContinuePaginatedPlan`.
+
+```mermaid
+classDiagram
+  direction LR
+  class ContinuePaginatedPlan {
+    <<AbstractPlan>>
+    -String cursor
+    -PlanSerializer planSerializer
+    -QueryService queryService
+  }
+```
+
+`ContinuePaginatedPlan` translated to entire Physical Plan Tree by `PlanSerializer` on cursor deserialization. It bypasses Logical Plan tree stage, `Planner`, `Optimizer` and `Implementor`.
+
+The examples below show Abstract Plan Tree for the same query in different request types:
+
+```mermaid
+stateDiagram-v2
+  state "Non Paged Request" as NonPaged {
+    state "QueryPlan" as QueryPlanNP
+    state "Project" as ProjectNP
+    state "Limit" as LimitNP
+    state "Filter" as FilterNP
+    state "Aggregation" as AggregationNP
+    state "Relation" as RelationNP
+
+    QueryPlanNP --> ProjectNP
+    ProjectNP --> LimitNP
+    LimitNP --> FilterNP
+    FilterNP --> AggregationNP
+    AggregationNP --> RelationNP
+  }
+
+  state "Initial Query Request" as Paged {
+    state "QueryPlan" as QueryPlanIP
+    state "Project" as ProjectIP
+    state "Limit" as LimitIP
+    state "Filter" as FilterIP
+    state "Aggregation" as AggregationIP
+    state "Relation" as RelationIP
+
+    Paginate --> QueryPlanIP
+    QueryPlanIP --> ProjectIP
+    ProjectIP --> LimitIP
+    LimitIP --> FilterIP
+    FilterIP --> AggregationIP
+    AggregationIP --> RelationIP
+  }
+
+  state "Subsequent Query Request" As Sub {
+    ContinuePaginatedPlan
+  }
+```
+
+#### Logical Plan tree
+
+Changes to plan tree for Initial Query Request with pagination:
+1. `LogicalPaginate` is added to the top of the tree. It stores information about paging/scrolling should be done in a private field `pageSize` being pushed down in the `Optimizer`.
```

Comment:
What is the `1.` for?

Reply:
There is a list of changes. Luckly, it has only one entry.